### ISSUE TITLE
[Serverless]: Update the semantic search section in the docs

### ISSUE THREAD

**Issue by szabosteve** [elastic/docs-content#61]
### Serverless Docs

Elasticsearch

### Description

1. Update the [semantic search](https://www.elastic.co/docs/current/serverless/elasticsearch/elasticsearch/reference/semantic-search) page under the Search your data section. A lot has happened in the semantic search domain since this page was updated, the latest changes are available in the classic docs.

2. Add a `semantic_text` tutorial next to the ELSER tutorial as a subpage of the semantic search page.

### Resources and additional context

Source of information:
* [semantic search page in the classic docs with the latest content](https://www.elastic.co/guide/en/elasticsearch/reference/current/semantic-search.html)
* [semantic_text tutorial in the classic docs](https://www.elastic.co/guide/en/elasticsearch/reference/current/semantic-search-semantic-text.html)

**Comment by kosabogi** [comment]
The documentation team will be making a decision this week on our approach to handling cases where documentation would potentially be duplicated on serverless. We are considering whether to maintain deployment-specific guides or consolidate content to avoid redundancy. Once a decision is made, I'll proceed with the necessary updates and move this issue forward accordingly.

**Comment by kosabogi** [comment]
To avoid duplication, we decided to remove redundant content from the Serverless documentation and add a page with links and descriptions for existing tutorials in the Elasticsearch documentation.

After restructuring, the Semantic search page now has the following structure:

- Semantic search: Introduction, description, a list of recommended workflows, and a link to the guide in the Elasticsearch documentation.
- Semantic search with the inference API workflow: Introduction, description, and a link to the guide in the Elasticsearch documentation.
- Semantic search with semantic_text workflow: Introduction, description, and a link to the guide in the Elasticsearch documentation.
- Semantic search with a deployed model workflow: Introduction, description, and a link to the guide in the Elasticsearch documentation.

Updated documentation: [Semantic search](https://www.elastic.co/guide/en/serverless/current/elasticsearch-reference-semantic-search.html)
